src/components/task-app/task-app.component.js

```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+import TaskForm from "../task-form";
+import TaskList from "../task-list";
+
+const TaskApp = () => {
+  const [toggleForm, setToggleForm] = useState(false);
+  const [taskList, setTaskList] = useState([]);
+
+  const openForm = () => {
+    setToggleForm(true);
+  };
+
+  const closeForm = () => {
+    setToggleForm(false);
+  };
+
+  const addNewTask = (data) => {
+    setTaskList([...taskList, data]);
```

Comment:
as mentioned in a previous comment (https://github.com/Joao-Matias/to-do-list/pull/2#discussion_r1129205073):

> a rule of thumb, whenever you're updating a state that relies on its previous value, use the callback method. if you're setting a new value (so you don't care about the previous state) you can directly pass the new value.

In this case, I would suggest updating this as:
```typescript
    setTaskList(previous => [...previous, data]);
```

Reply:
277c96714382536b62de549c29ffe9a09aa6ac9f Changed to previous state as advised above.